### PR TITLE
refactor rlx message

### PR DESCRIPTION
**Pull request by genywind**

### CODE REVIEW

File: relaax/common/rlx_message.py [hunk 1]
```diff
@@ -21,166 +21,185 @@ class RLXMessage(object):
     TYPE_LIST = 8
     TYPE_UINT4 = 9
 
+    pack_info = {
```
Comment (StanislavVolodarskiy) [inline relaax/common/rlx_message.py]: why do you use type names and dict keys? Types itself are OK as keys.

File: relaax/common/rlx_message.py [hunk 2]
```diff
@@ -21,166 +21,185 @@ class RLXMessage(object):
...
 
+        bval = bytearray(str(value).encode('UTF-8'))
+        buf += pack("I", len(bval))
+        buf += bval
     @classmethod
     def _unpack_string(cls, buf, offset):
         reslen = unpack_from("I", buf, offset)[0]
         offset += 4
-        res = str(buf[offset:offset+reslen].decode('UTF-8'))
+        test = buf[offset:offset+reslen]
```
Comment (StanislavVolodarskiy) [inline relaax/common/rlx_message.py]: 'test' is strange name.

File: relaax/common/rlx_message.py [hunk 3]
```diff
@@ -21,166 +21,185 @@ class RLXMessage(object):
...
+        return res, offset
+
+    @classmethod
+    def _pack_bool(cls, value, buf, pack_type=True):
+        cls._pack_type(cls.TYPE_BOOLEAN, buf, pack_type)
+        buf += pack("B", 1 if value else 0)
+    @classmethod
+    def _unpack_bool(cls, buf, offset):
+        res = unpack_from("B", buf, offset)[0]
+        res = True if res == 1 else False
```
Comment (StanislavVolodarskiy) [inline relaax/common/rlx_message.py]: res = res == 1

File: relaax/common/rlx_message.py [hunk 4]
```diff
@@ -21,166 +21,185 @@ class RLXMessage(object):
...
-            res = res.reshape(shape)
-            offset += reslen
-        elif valtype == cls.TYPE_LIST:
-            reslen = unpack_from("I", buf, offset)[0]
+        bval = value.image.tobytes()
+        buf += pack("I", len(bval))
+        buf += bval
+    @classmethod
+    def _unpack_image(cls, buf, offset):
+        (mode, offset) = cls._unpack_string(buf, offset)
```
Comment (StanislavVolodarskiy) [inline relaax/common/rlx_message.py]: parens are not neccesary.

File: relaax/common/rlx_message.py [hunk 5]
```diff
@@ -21,166 +21,185 @@ class RLXMessage(object):
...
+        x = unpack_from("I", buf, offset)[0]
+        offset += 4
+        y = unpack_from("I", buf, offset)[0]
+        offset += 4
+
+        reslen = unpack_from("I", buf, offset)[0]
+        offset += 4
+        img = Image.frombytes(mode, (x, y), bytes(buf[offset:offset+reslen])).convert("RGB")
+        res = np.asarray(img)
```
Comment (StanislavVolodarskiy) [inline relaax/common/rlx_message.py:132]: We send image and receive numpy.array. Strange for me.

File: relaax/common/rlx_message.py [hunk 6]
```diff
@@ -21,166 +21,185 @@ class RLXMessage(object):
...
-                res.append(item)
-        else:
-            print("Unknown type:%d" % valtype)
-            raise Exception("Unknown type:%d" % valtype)
+            shape.append(item)
 
+        reslen = unpack_from("I", buf, offset)[0]
+        offset += 4
+        res = np.frombuffer(buf[offset:offset+reslen], dtype=np.dtype(dtype))
+        res = res.reshape(shape)
```
Comment (StanislavVolodarskiy) [inline relaax/common/rlx_message.py]: res = np.ndarray(shape=shape, dtype=np.dtype(dtype), buffer=buf[offset:offset+reslen])

File: tests/rlx_message_test.py [hunk 7]
```diff
@@ -17,10 +17,11 @@ def test_to_wire_and_back(self):
...
-        data = {'arg1': 1, 'arg_1': -1, "args": "test1", "argn": -234, "argbt": True, "argbf": False, "argnone": None,
+        data = {'arg1': 1, 'arg_1': -1, "args": "test1", "argn": -234, "argd": 0.067, "argbt": True, "argbf": False, "argnone": None,
                 "array_empty": [], 'array': [4.02, 5.2, 6.006],
                 'nparray1': npar1, 'nparray2': npar2, 'nparray1empty': npar1empty, 'nparray2empty': npar2empty,
-                'image_jpg': RLXMessageImage(imageJPG), 'image_bmp': RLXMessageImage(imageBMP)}
+                'image_jpg': RLXMessageImage(imageJPG), 'image_bmp': RLXMessageImage(imageBMP)
+                }
         wire = RLXMessage.to_wire(data)
         back = RLXMessage.from_wire(wire)
         assert len(data) == len(back)
```
Comment (StanislavVolodarskiy) [inline tests/rlx_message_test.py:28]: assert doesn't check equality on content of messages.

File: relaax/common/rlx_message.py (hunk 5 above)
Comment (genywind) [reply relaax/common/rlx_message.py:132]: It is by design

File: tests/rlx_message_test.py (hunk 7 above)
Comment (genywind) [reply tests/rlx_message_test.py:28]: see code below

File: relaax/common/rlx_message.py (hunk 5 above)
Comment (StanislavVolodarskiy) [reply relaax/common/rlx_message.py:132]: Ok.

**Pull request merged**